tests/handlers/test_directory.py

```diff
@@ -458,20 +458,32 @@ def prepare(self, reactor, clock, hs):
         return hs
 
     def test_denied(self):
```

Comment:
Are these helper functions or separate tests (or both?) It looks like they're helpers, but by naming them `test_...` they'll get run as individual tests also.

A docstring might clarify this?